src/Microsoft.AspNet.Mvc.ViewFeatures/ViewEngines/CompositeViewEngine.cs

```diff
@@ -23,62 +26,69 @@ public CompositeViewEngine(IOptions<MvcViewOptions> optionsAccessor)
         /// <inheritdoc />
         public IReadOnlyList<IViewEngine> ViewEngines { get; }
 
-        /// <inheritdoc />
-        public ViewEngineResult FindPartialView(
-            ActionContext context,
-            string partialViewName)
+        /// <summary>
+        /// Determine if given name should be interpreted as a page path.
+        /// </summary>
+        /// <param name="name">Name of or path to a page.</param>
+        /// <returns><c>true</c> if the given <paramref name="name"/> appears to be a path to a page.</returns>
+        public static bool IsPagePath(string name)
         {
-            if (context == null)
-            {
-                throw new ArgumentNullException(nameof(context));
-            }
-
-            if (partialViewName == null)
-            {
-                throw new ArgumentNullException(nameof(partialViewName));
-            }
+            Debug.Assert(!string.IsNullOrEmpty(name));
 
-            return FindView(context, partialViewName, partial: true);
+            // A page path starts with "~" or "/" (if absolute) or ends with ".cshtml" (if relative).
+            return name[0] == '~' ||
+                name[0] == '/' ||
+                name.EndsWith(ViewExtension, StringComparison.OrdinalIgnoreCase);
```

Comment:
If @Eilon agrees w/ this concern, `IViewEngine` would need to expose `IsPagePath()` and this class would determine truthiness based on its contained `IViewEngine` implementations. E.g. first `IViewEngine` that says `true` determines the return value.

Of course this is perilous because the view engine that said something was a path (and not a name) will not necessarily be called first when `GetView()` is called. Might need a `IViewEngine HandleAsPath(string path)` method on `IViewEngine` instead.